salt/_grains/etcd_info.py

```diff
@@ -0,0 +1,68 @@
+import os
+# Import python libs
+import logging
+
+# Import third party libs
+try:
+    import etcd
+    HAS_LIBS = True
+except ImportError:
+    HAS_LIBS = False
+
+# Set up logging
+log = logging.getLogger(__name__)
+
+def etcd_info():
+  # TODO: Certificates configuration should be 
+  # provided in pillar profile config
+  ca_cert = "/etc/pki/trust/anchors/SUSE_CaaSP_CA.crt"
+  cli_cert = "/etc/pki/minion.crt"
+  cli_key = "/etc/pki/minion.key"
+
+  certs = {}
+  if (os.path.isfile(ca_cert) and 
+      os.path.isfile(cli_cert) and 
+      os.path.isfile(cli_key)):
+    cert = (cli_cert, cli_key)
+    certs = {
+      'ca_cert': ca_cert,
+      'cert': cert,
+      'protocol': 'https'
+    }
+
+  etcd_info = {}
+  etcd_info['members_all'] = []
+  etcd_info['member_type'] = ""
+
+  xargs = certs.copy()
```

Comment:
I think you don't need this copy. You could even use a more compact form with something like:

```
client_args = {}
if os.path.isfile(ca_cert): client_args['ca_cert'] = ca_cert
if os.path.isfile(cli_key) and os.path.isfile(cli_cert):
  client_args.update({'cert': (cli_cert, cli_key),  'protocol': 'https'})
...
client = etcd.Client(host='localhost', port=2379, **client_args)
```